unity/Assets/Maroon/scenes/experiments/MinimumSpanningTree/Scripts/IslandSetUp.cs

```diff
@@ -0,0 +1,242 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+public class IslandSetUp : MonoBehaviour, IResetObject
+{
+    public GameObject islandPrefab01;
+    public GameObject islandPrefab02;
+    public GameObject islandPrefab03;
+    public GameObject islandPrefab04;
+    
+    public Transform topLeftBorder;
+    public Transform bottomRightBorder;
+
+    static Vector3 radius;
+    int _numberOfIslands = 4;
+
+    protected List<GameObject> islandClones = new List<GameObject>();
+
+    //public LayerMask m_LayerMask;
+
+    private void Awake()
+    {
+        //Debug.Log("IslandSetUp Awake");
+    }
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        //Debug.Log("IslandSetUp Start");
+        // 1.9f for a little extra space
+        radius = islandPrefab01.GetComponent<MeshRenderer>().bounds.size / 1.9f;
+        createIslands();
+    }
+
+    /**
+     * Set a Random Position within Range
+     * */
+    Vector3 setIslandPosition()
```

Comment:
Please follow the c# coding conventions
https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions

e.g. SetIslandPosition